code/game/atoms.dm

```diff
@@ -523,6 +523,12 @@
 		if("color")
 			add_atom_colour(color, ADMIN_COLOUR_PRIORITY)
 
+		if("dir")
+			var/do_setdir = alert(usr, "use setDir?", src, "Yes", "No")
+			if(do_setdir == "Yes")
+				setDir(var_value)

```

Comment:
Calling setDir after the value has already changed wont play nicely with functionality which expect the existing dir value to be the old one.